src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
+        Random random = new Random();
+        int figureNumber = random.nextInt(5);
```

Comment:
Consider defining the number of figures as a constant, as suggested in the checklist: "All magic numbers in your code should be constants." This will improve code readability and maintainability.